components/quick_switch_modal/quick_switch_modal.tsx

```diff
@@ -17,33 +17,55 @@ import SuggestionBox from 'components/suggestion/suggestion_box.jsx';
 import SuggestionList from 'components/suggestion/suggestion_list.jsx';
 import SwitchChannelProvider from 'components/suggestion/switch_channel_provider.jsx';
 import SwitchTeamProvider from 'components/suggestion/switch_team_provider.jsx';
-import NoResultsIndicator from 'components/no_results_indicator/no_results_indicator.tsx';
+import NoResultsIndicator from 'components/no_results_indicator/no_results_indicator';
 
 import {NoResultsVariant} from 'components/no_results_indicator/types';
 
 const CHANNEL_MODE = 'channel';
 const TEAM_MODE = 'team';
 
-export default class QuickSwitchModal extends React.PureComponent {
-    static propTypes = {
+type SwitchToChannelResult = {
+    data?: true;
+    error?: true;
+}
+type ProviderSuggestions = {
+    matchedPretext: any,
+    terms: string[],
+    items: any[],
+    component: SwitchChannelProvider|SwitchTeamProvider,
```

Comment:
The provider isn't a component itself, but it returns its own component. I'm pretty sure this can just be 
```suggestion
    component: React.ReactNode,
```
so that it accepts and component.